migrate: supports INSERT Statement

knwoop opened the pull request:
<!--
Please read the contribution guidelines and the CLA carefully before
submitting your pull request.

https://cla.developers.google.com/
-->

## WHAT

- support INSERT statement.

## WHY

- to support INSERT statement.
- Currently, Cloud Spanner does not support INSERT in partitioned DML (INSERT was treated as partitioned DML in wrench)
  - ref: https://cloud.google.com/spanner/docs/dml-partitioned#unsupported-features

shuheiktgw commented on pkg/spanner/migration.go:
```diff
@@ -171,29 +173,46 @@ func dmlToStatements(filename string, data []byte) ([]string, error) {
 
 func inspectStatementsKind(statements []string) (statementKind, error) {
```
nit, The current code looks unnecessarily complicated to me. How about refactoring it like below?

```go
func inspectStatementsKind(statements []string) (statementKind, error) {
	var hasDDL, hasDML, hasPartitionedDML bool
	for _, s := range statements {
		switch {
		case isDML(s):
			hasDML = true
		case isPartitionedDML(s):
			hasPartitionedDML = true
		default:
			hasDDL = true
		}
	}

	switch {
	case hasDDL && !hasDML && !hasPartitionedDML:
		return statementKindDDL, nil
	case !hasDDL && hasDML && !hasPartitionedDML:
		return statementKindDML, nil
	case !hasDDL && !hasDML && hasPartitionedDML:
		return statementKindPartitionedDML, nil
	default:
		return "", errors.New("DDL, DML (INSERT), and partitioned DML (UPDATE or DELETE) must not be combined in the same migration file")
	}
}
```

shuheiktgw reviewed (commented):
Thank you for your PR, @knwoop, sorry that we could not get back to you sooner 🙇 
I left one minor comment, but except that it looks good to me!

knwoop commented:
> Thank you for your PR, @knwoop, and sorry that we could not get back to you sooner 🙇 I left one minor comment, but except that it looks good to me!

@shuheiktgw Thank you for reviewing my PR! I appreciate your feedback and have addressed the comment.

knwoop commented on pkg/spanner/migration.go:
```diff
@@ -170,30 +172,38 @@ func dmlToStatements(filename string, data []byte) ([]string, error) {
-	kindMap := map[statementKind]uint64{
-		statementKindDDL: 0,
-		statementKindDML: 0,
+	if len(statements) == 0 { // Treat empty files as DDL.
```
Note: this is treated as a DDL under the existing specifications.

shuheiktgw reviewed (approved):
LGTM! I'll ask for a review from another engineer as well.

utahta reviewed (approved):
LGTM!
